src/grammar.coffee

```diff
@@ -660,26 +660,31 @@ grammar =
   # Comprehensions can either be normal, with a block of expressions to execute,
   # or postfix, with a single expression.
   For: [
-    o 'Statement  ForBody',                     -> new For $1, $2
-    o 'Expression ForBody',                     -> new For $1, $2
-    o 'ForBody    Block',                       -> new For $2, $1
-    o 'ForLineBody  Block',                     -> new For $2, $1
+    o 'Statement    ForBody',  -> $2.addBody $1
+    o 'Expression   ForBody',  -> $2.addBody $1
+    o 'ForBody      Block',    -> $1.addBody $2
+    o 'ForLineBody  Block',    -> $1.addBody $2
```

Comment:
Can you please explain all these grammar changes and why they’re necessary (or an improvement)?

Reply:
@vendethiel suggested moving some code from `grammar` into `nodes`, e.g.

```coffeescript
 o 'ForStart ForSource',  ->
        $2[prop] = $1[prop] for prop in ['await', 'awaitTag', 'own', 'ownTag']
```

to which I agreed, and then reworked `For` in `nodes`.
It's just a cleanup.

Reply:
Ah, okay. Maybe next time that should be its own PR, so it’s easier to track the changes that relate to the cleanup as opposed to adding support for async iterators, but I have no objection.